ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/rlpx/RlpxAgent.java

```diff
@@ -383,52 +407,83 @@ && getConnectionCount() >= maxConnections) {
       return;
     }
 
+    peerConnection.setOnConnectionReadyCallback(() -> callOnConnectionReady(peerConnection, true));
+    dispatchConnect(peerConnection);
+  }
+
+  @VisibleForTesting
+  private boolean callOnConnectionReady(
+      final PeerConnection peerConnection, final boolean isInboundConnection) {
+
     // Track this new connection, deduplicating existing connection if necessary
     final AtomicBoolean newConnectionAccepted = new AtomicBoolean(false);
-    final RlpxConnection inboundConnection = RlpxConnection.inboundConnection(peerConnection);
+    final Peer peer = peerConnection.getPeer();
+    final RlpxConnection newConnection =
+        isInboundConnection
+            ? RlpxConnection.inboundConnection(peerConnection)
+            : RlpxConnection.outboundConnection(
+                peer, CompletableFuture.completedFuture(peerConnection));
+
+    // Disconnect if too many peers
+
+    if (!randomPeerPriority) {
```

Comment:
I have the feeling that all these checks are done when the connection comes from a peer. so do you think if we should check isInboundConnection to know if we need to do this part